sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +310,102 @@ export class ManagementClient extends LinkEntity {
     options?: { retryOptions?: RetryOptions; abortSignal?: AbortSignalLike; requestName?: string }
   ): Promise<any> {
     try {
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
-
       if (!options) {
         options = {};
       }
 
-      const sendRequestOptions: SendRequestOptions = {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            let count = 0;
+
+            if (!options) {
+              options = {};
+            }
+
+            const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
+            let timeTakenByInit = 0;
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
```

Comment:
We should check if the operation has been cancelled before this point so we don't do an unnecessary initialization.

Reply:
We are doing that check right before this line, that should suffice right?

Reply:
Prior to this comment, I had removed abort for around init altogether, since the test "respects cancellation token" was failing.
This was because init doesn't accept the abortSignal and so the connection, sessions, sender/receiver links and associated listeners, timers were all still staying open. 
I had tried a lot of things, explicitly closing, clearing them, but looks like the connection needs to be explicitly closed and doing that was not trivial as reference to connection within abandoned init was getting lost. See https://github.com/Azure/azure-sdk-for-js/pull/4322#discussion_r307429438 

Here, I agreed with making the synchronous check for before the init for in case we know by then that operation was aborted.